_posts/2024-09-16-introducing-opensearch-2-17.md

```diff
@@ -0,0 +1,109 @@
+
+## Search
+
+This release also delivers new functionality focused on helping you improve search and query performance.
+
+**Optimize search performance with enhanced concurrent segment search settings and expanded support**
+
+[Concurrent segment search](https://opensearch.org/docs/latest/search-plugins/concurrent-segment-search/) has been generally available since OpenSearch 2.12. It supports two tuning parameters: an/off switch and a cluster-level setting that specifies the number of slices (maximum thread count) per request. In general, a greater number of slices reduces latency but requires more processing power per request. [More specific analysis](https://opensearch.org/blog/concurrent-search-follow-up/) has shown that some request types (including aggregations and remote store searches) benefit more from concurrent segment search than others. To help you take advantage of concurrent segment search for the right request types, OpenSearch 2.17 introduces a new setting both at index and cluster level. These settings along with [pluggable _decider_ logic](https://github.com/opensearch-project/OpenSearch/pull/15363) provides more granular control over which requests use concurrent search. This release also supports a [per-index (as opposed to per-cluster) maximum slice count setting](https://github.com/opensearch-project/OpenSearch/pull/15336). Additionally, we have expanded concurrent segment search to support more script-based search requests, including [composite aggregations with scripting](https://github.com/opensearch-project/OpenSearch/pull/15072) and [derived fields](https://github.com/opensearch-project/OpenSearch/pull/15326).
+
+**Enhance query performance with efficient numeric term encoding and roaring bitmaps**
+
+The `terms` query type has been provided by OpenSearch since the beginning, serving as a method to filter searches by including documents that contain a field value within a specified set of values. It has always supported a _terms lookup_, which retrieves those filter values from another index based on a document ID, essentially allowing a limited version of cross-index join. A limitation has been that the number of values retrieved from the lookup index was limited to the hundreds (or potentially low thousands). In OpenSearch 2.17, we have [added support for encoding numeric term values](https://github.com/opensearch-project/OpenSearch/pull/14774) as a [roaring bitmap](https://roaringbitmap.org/). By encoding the values more efficiently, a search request can use a stored filter that matches over a million documents, with lower retrieval latency and less memory used. For details, see [Bitmap filtering](https://opensearch.org/docs/latest/query-dsl/term/terms/#bitmap-filtering).
```

Comment:
```suggestion
OpenSearch has provided the `terms` query type since the beginning, serving as a method to filter searches by including documents that contain a field value within a specified set of values. It has always supported a _terms lookup_, which retrieves those filter values from another index based on a document ID, essentially allowing a limited version of cross-index join. A previous limitation has been that the number of values retrieved from the lookup index was limited to the hundreds (or potentially low thousands). In OpenSearch 2.17, we have [added support for encoding numeric term values](https://github.com/opensearch-project/OpenSearch/pull/14774) as a [roaring bitmap](https://roaringbitmap.org/). By encoding the values more efficiently, a search request can use a stored filter that matches over a million documents, with lower retrieval latency and less memory used. For more information, see [Bitmap filtering](https://opensearch.org/docs/latest/query-dsl/term/terms/#bitmap-filtering).
```